herbert_core/utilities/maths/noisify.m

```diff
@@ -17,40 +17,69 @@
 %           Add noise with Poisson distribution, where the mean value at
 %           a point is the value y.
 %
-%   If no input errors, e, just set e=[]
+%   >> [yout, eout] = noisify(y,e,[factor,]'maximum_value',maxval)
+%           Add noise with Gaussian distribution, calculating the standard
+%           deviation by an externally provided maximum y value.
+%           The max value is preceded by a keyword string 'maximum_value'.
+%           Typically this value will be the overall maximum value if the
+%           data is processed page by page and the maximum must be
+%           extracted before this processing.
+%           Omitting factor will use the default value 0.1.
+%
+%   If no input errors, e, just set e=[]. Note that eout will be created
+%   regardless of whether e is present, so the first overload output is
+%   probably incomplete.
+
+% deal with optional arguments
+is_poisson= false;
+fac = 0.1;
+if nargin>=2
+    p = inputParser;
+    addRequired(p,'y');   % y compulsory
+    addRequired(p,'e');   % e compulsory if any of remaining optional/parameter arguments present
```

Comment:
Should we have a `@isnumeric` validator here?

Reply:
`e` is optional, but positionally significant. If you define it as required does the parser raise an error when it's missing?

Reply:
e is required at that point in the code as it is within a nargin>=2 block, so it will never be missing as coded. Having e and disp_or_fac as both optional is problematic, hence the restriction. I tried isnumeric(y) in the debugger and it worked so I could add that.

Reply:
Why is it problematic if `e` and `disp_or_fac` are both optional?

Reply:
As said above, you could leave out e and include disp_or_fac. I don't see anything in the inputParser docs which suggest this is impossible - it seems that ignoring it is done partly on the basis of the validation function argument, which will skip the argument slot if not validated and move on to the next one.